scipy/constants/constants.py

```diff
@@ -173,6 +173,75 @@
 # functions for conversions that are not linear
 
 
+def convert_temperature(val, old_scale, new_scale):
+    """
+    Convert from a temperature scale to another one among Celsius, Kelvin,
+    Fahrenheit and Rankine scales.
+
+    Parameters
+    ----------
+    val : array_like
+        Value(s) of the temperature(s) to be converted expressed in the
+        original scale.
+
+    old_scale: str
+        Specifies as a string the original scale from which the temperature
+        value(s) will be converted. Supported scales are Celsius ('Celsius',
+        'celsius', 'C' or 'c'), Kelvin ('Kelvin', 'kelvin', 'K', 'k'),
+        Fahrenheit ('Fahrenheit', 'fahrenheit', 'F' or 'f') and Rankine
+        ('Rankine', 'rankine', 'R', 'r').
+
+    new_scale: str
+        Specifies as a string the new scale to which the temperature
+        value(s) will be converted. Supported scales are Celsius ('Celsius',
+        'celsius', 'C' or 'c'), Kelvin ('Kelvin', 'kelvin', 'K', 'k'),
+        Fahrenheit ('Fahrenheit', 'fahrenheit', 'F' or 'f') and Rankine
+        ('Rankine', 'rankine', 'R', 'r').
+
+    Returns
+    -------
+    res : float or array of floats
+        Value(s) of the converted temperature(s) expressed in the new scale.
+
+    Examples
+    --------
+    >>> from scipy.constants import convert_temperature
+    >>> convert_temperature(np.array([-40, 40.0]), 'Celsius', 'Kelvin')
+    array([ 233.15,  313.15])
+
+    """
+    if old_scale.lower() in ['celsius', 'c']:
```

Comment:
Maybe add ``# Convert from `old_scale` to kelvin``
and later on ``# and from kelvin to `new_scale`.``